src/utils/utils_common.h

```diff
@@ -29,8 +29,13 @@ extern "C" {
         expression;                                                            \
     } while (0)
 
-#define ALIGN_UP(value, align) (((value) + (align)-1) & ~((align)-1))
-#define ALIGN_DOWN(value, align) ((value) & ~((align)-1))
+#define ALIGN_UP(value, align) ((value + align - 1) & ~(align - 1))
+#define ALIGN_UP_SAFE(value, alignment)                                        \
+    ((alignment == 0) ? (value)                                                \
+                      : ((value + alignment - 1) < value                       \
+                             ? 0                                               \
+                             : ((value + alignment - 1) & ~(alignment - 1))))
+#define ALIGN_DOWN(value, align) (value & ~(align - 1))
```

Comment:
It should stay as was:
```c
#define ALIGN_DOWN(value, align) ((value) & ~((align)-1))
```
Please revert it.

Reply:
Reverted